### PR TITLE
Update 'prepare_managed_clusters_data' to pass with args set to None value

### PR DESCRIPTION
**Pull request by oharan2**
When gcp_service_account_file is None, `get_service_account_dict_from_file` fails when unneeded (for example a ROSA cluster)

Prow testing failure example:
`TypeError: expected str, bytes or os.PathLike object, not NoneType`
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/43065/rehearse-43065-pull-ci-rhpit-interop-tests-main-rosa-sts-test-rhods-operator-4.13-lp-rosa-rosa-sts-rhods-test-operator/1704533939206492160#1:build-log.txt%3A114

### CODE REVIEW

File: openshift_cli_installer/libs/managed_clusters/helpers.py [hunk 1]
```diff
@@ -41,6 +40,17 @@ def prepare_managed_clusters_data(
                 _cluster["machine-cidr"] = _cluster.get("cidr", "10.0.0.0/16")
 
         elif cluster_platform == GCP_OSD_STR:
+            if gcp_service_account_file:
```
Comment (rnetser) [inline openshift_cli_installer/libs/managed_clusters/helpers.py:43]: please remove, this is checked in https://github.com/RedHatQE/openshift-cli-installer/blob/main/openshift_cli_installer/utils/cli_utils.py#L472

File: openshift_cli_installer/libs/managed_clusters/helpers.py [hunk 2]
```diff
@@ -41,6 +40,17 @@ def prepare_managed_clusters_data(
...
+                gcp_service_account_dict = get_service_account_dict_from_file(
+                    gcp_service_account_file=gcp_service_account_file
+                )
+            else:
+                click.secho(
+                    "'gcp_service_account_file' wasn't provided, but cluster_platform"
+                    " is gcp-osd",
+                    fg=ERROR_LOG_COLOR,
+                )
+                raise click.Abort()
```
Comment (rnetser) [inline openshift_cli_installer/libs/managed_clusters/helpers.py:53]: ditto

**Review by rnetser — CHANGES_REQUESTED** [review]

File: openshift_cli_installer/libs/managed_clusters/helpers.py [hunk 3]
```diff
@@ -16,12 +19,8 @@ def prepare_managed_clusters_data(
     aws_account_id,
     aws_secret_access_key,
     aws_access_key_id,
```
[the remark this answers is not in the document]
Comment (oharan2) [reply openshift_cli_installer/libs/managed_clusters/helpers.py]: Done

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by oharan2** [comment]
Resolved in #190

**Pull request closed without merging**